clang/lib/CIR/CodeGen/CIRGenFunction.h

```diff
@@ -105,6 +106,27 @@ class CIRGenFunction : public CIRGenTypeCache {
   mlir::Value emitAlloca(llvm::StringRef name, mlir::Type ty,
                          mlir::Location loc, clang::CharUnits alignment);
 
+private:
+  // Track current variable initialization (if there's one)
+  const clang::VarDecl *currVarDecl = nullptr;
+  class VarDeclContext {
+    CIRGenFunction &p;
+    const clang::VarDecl *oldVal = nullptr;
+
+  public:
+    VarDeclContext(CIRGenFunction &p, const VarDecl *value) : p(p) {
+      if (p.currVarDecl)
```

Comment:
SO I was going through debugging and found that this is somewhat odd/looks strange to me.  Is there a good reason that we have this guarded to set `oldVal`, but unconditionally restore it?  I realize this ends up 'working' because `currVarDecl` is a pointer (and `oldVal` is initialized to a `nullptr`), thus this ends up working, but this condition seems REALLY strange.  It is at best confusing/unnecessary, at worst, a bug in `restore`.